sys/include/libufs.h

```diff
@@ -10,11 +10,22 @@
 
 
 struct Chan;
-struct mount;
 struct vnode;
+struct thread;
 
+/* Wrapper for a UFS mount.  Should support reading from both kernel and user
+ * space (eventually)
+ */
+struct mount {
+	struct ufsmount*	mnt_data;
+	struct Chan*		chan;
+	int32_t			(*read)(struct mount*, void*, int32_t, int64_t);
```

Comment:
I'm trying to make the library usable for kernel or user code.  I could only see a way to get access to the channel's internal dev member (in portdat.h) in the kernel code.  Also, I assumed people would usually open a file outside the kernel using 'open'.  This way permits either, assuming we eventually make the Chan a union with a fid (or do something else).